editor/hooks/autocomplete/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { compose } from 'lodash';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Autocomplete, withContext } from '@wordpress/components';
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { blockAutocompleter, userAutocompleter } from './autocompleters';
+import { getInserterItems } from '../../store/selectors';
+
+function withAutocomplete( BlockEdit ) {
+	function WrappedBlockEdit( props ) {
+		const { name, items, onReplace } = props;
+
+		// Only wrap paragraph blocks with <Autocomplete>
+		if ( name !== 'core/paragraph' ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		return (
+			<Autocomplete completers={ [ blockAutocompleter( { items, onReplace } ), userAutocompleter() ] }>
+				{ ( { isExpanded, listBoxId, activeId } ) => (
+					// TODO: These aria attributes probably belong on the <Editable>... not sure how to make that happen
+					// eslint-disable-next-line jsx-a11y/aria-activedescendant-has-tabindex
+					<div
+						aria-autocomplete="list"
+						aria-expanded={ isExpanded }
+						aria-owns={ listBoxId }
+						aria-activedescendant={ activeId }
+					>
+						{ <BlockEdit { ...props } /> }
+					</div>
+				) }
+			</Autocomplete>
+		);
+	}
+
+	return compose(
+		withContext( 'editor' )( settings => ( {
+			enabledBlockTypes: settings.blockTypes,
+		} ) ),
+		connect( ( state, ownProps ) => ( {
+			items: getInserterItems( state, ownProps.enabledBlockTypes ),
```

Comment:
As part of integrating Gutenberg into Automattic P2's, we need to use a different data source for the users autocompleter because we allow pinging any Automattician, not just users who are a member of a particular P2. I'm not too familiar with how data dependencies are satisfied within Gutenberg (except my perception is that there are a number of approaches and that `withApiData` is preferred for the future). Do you see a good way for us to override the request and provision of completion data?

Reply:
I didn't read well enough :). I see now that only `blockAutocompleter` is initialized with a list of items. Other completers like the `userAutocompleter` could source their own data if they want.